packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -252,12 +252,16 @@ class FlutterDevice {
             ipv6,
             disableServiceAuthCodes,
           );
-        } on dds.DartDevelopmentServiceException catch (e) {
+        } on dds.DartDevelopmentServiceException catch (e, st) {
           if (!allowExistingDdsInstance ||
               (e.errorCode != dds.DartDevelopmentServiceException.existingDdsInstanceError)) {
             handleError(e);
             return;
           } else {
+            if (device.dds.uri == null) {
+              completer.completeError(e, st);
```

Comment:
Now that you've written this to plumb the stack trace, I'm noticing that `handleError()` should also get the stack trace to pass it when it calls `completeError()`.